core/src/services/tikv/backend.rs

```diff
@@ -0,0 +1,257 @@
+        }
+        self
+    }
+}
+
+impl Builder for TiKVBuilder {
+    const SCHEME: Scheme = Scheme::Redb;
+    type Accessor = Backend;
+
+    fn from_map(map: HashMap<String, String>) -> Self {
+        let mut builder = TiKVBuilder::default();
+
+        map.get("endpoints")
+            .map(|v| v.split(',').map(|s| s.to_owned()).collect::<Vec<String>>())
+            .map(|v| builder.endpoints(v));
+        map.get("insecure")
+            .filter(|v| *v == "on" || *v == "true")
+            .map(|_| builder.insecure());
+        map.get("ca_path").map(|v| builder.ca_path(v));
+        map.get("cert_path").map(|v| builder.cert_path(v));
+        map.get("key_path").map(|v| builder.key_path(v));
+
+        builder
+    }
+
+    fn build(&mut self) -> Result<Self::Accessor> {
+        let endpoints = self.endpoints.take().ok_or_else(|| {
+            Error::new(
+                ErrorKind::ConfigInvalid,
+                "endpoints is required but not set",
+            )
+            .with_context("service", Scheme::TiKV)
+        })?;
+
+        if self.insecure {
+            if self.ca_path.is_some() || self.key_path.is_some() || self.cert_path.is_some() {
+                return Err(
+                    Error::new(ErrorKind::ConfigInvalid, "invalid tls configuration")
+                        .with_context("service", Scheme::TiKV)
+                        .with_context("endpoints", format!("{:?}", endpoints)),
+                )?;
+            }
+        } else if !(self.ca_path.is_some() && self.key_path.is_some() && self.cert_path.is_some()) {
+            return Err(
+                Error::new(ErrorKind::ConfigInvalid, "invalid tls configuration")
+                    .with_context("service", Scheme::TiKV)
+                    .with_context("endpoints", format!("{:?}", endpoints)),
+            )?;
+        }
+
+        Ok(Backend::new(Adapter {
+            client: None,
+            endpoints,
+            insecure: self.insecure,
+            ca_path: self.ca_path.clone(),
+            cert_path: self.cert_path.clone(),
+            key_path: self.key_path.clone(),
+        }))
+    }
+}
+
+/// Backend for TiKV service
+pub type Backend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    client: Option<RawClient>,
+    endpoints: Vec<String>,
+    insecure: bool,
+    ca_path: Option<String>,
+    cert_path: Option<String>,
+    key_path: Option<String>,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        let mut ds = f.debug_struct("Adapter");
+        ds.field("endpoints", &self.endpoints);
+        ds.field("insecure", &self.insecure);
+        ds.field("ca_path", &self.ca_path);
+        ds.field("cert_path", &self.cert_path);
+        ds.field("key_path", &self.key_path);
+        ds.finish()
+    }
+}
+
+impl Adapter {
+    async fn get_connection(&self) -> Result<RawClient> {
+        if let Some(client) = self.client.clone() {
```

Comment:
self.client will always be `None`.

Reply:
We can use `tokio::sync::OnceCell` here to make sure it init once.

Reply:
But use `tokio` base library will make `opendal` core more redundancy？